transparency/boot_entry.go

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) The go-boot authors. All Rights Reserved.
+//
+// Use of this source code is governed by the license
+// that can be found in the LICENSE file.
+
+// Package transparency implements an interface to the
+// boot-transparency library functions to ease boot bundle
+// validation.
+package transparency
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	"github.com/usbarmory/boot-transparency/artifact"
+	"github.com/usbarmory/boot-transparency/engine/sigsum"
+	"github.com/usbarmory/boot-transparency/policy"
+	"github.com/usbarmory/boot-transparency/transparency"
+)
+
+// Artifact represents a boot artifact.
+type Artifact struct {
+	// Category must be consistent with the artifact categories
+	// supported by boot-transparency library.
+	Category uint
+
+	// SHA-256 hash of the artifact.
+	Hash string
+}
+
+// BootEntry represent a boot antry as a set of artifacts.
+type BootEntry []Artifact
+
+// Validate the transparency inclusion proof and, the consistency
+// between the boot policy and the logged claims for the boot artifacts.
+// Takes as input the pointer to the boot transparency configuration
+// to validate the boot entry artifacts.
+// Returns error if the boot artifacts are not passing the validation.
+func (b *BootEntry) Validate(c *Config) (err error) {
+	if c.Status == None {
+		return
+	}
+
+	if b == nil || len(*b) == 0 {
+		return fmt.Errorf("got an invalid boot entry pointer")
+	}
+
+	// Automatically load the configuration from the UEFI partition
+	// when the function is used within the UEFI boot loader.
+	if c.UefiRoot != nil {
+		entryPath, err := c.Path(b)
+		if err != nil {
+			return fmt.Errorf("cannot load boot transparency configuration, %v", err)
+		}
+
+		if err = c.loadFromUefiRoot(entryPath); err != nil {
+			return fmt.Errorf("cannot load boot transparency configuration, %v", err)
+		}
+	}
+
+	te, err := transparency.GetEngine(transparency.Sigsum)
+	if err != nil {
+		return fmt.Errorf("unable to configure the transparency engine, %w", err)
+	}
+
+	if err = te.SetKey([]string{string(c.LogKey)}, []string{string(c.SubmitKey)}); err != nil {
```

Comment:
Maybe `SetKey` should take `[]byte` as arguments given that typically keys in Go are defined as such, and this would ease this invocation as well.